docs/csharp/language-reference/keywords/value-types.md

```diff
@@ -9,26 +9,28 @@ helpviewer_keywords:
   - "C# language, value types"
 ms.assetid: 471eb994-2958-49d5-a6be-19b4313f80a3
 ---
-# Value Types (C# Reference)
-The value types consist of two main categories:  
-  
--   [Structs](../../../csharp/language-reference/keywords/struct.md)  
-  
--   [Enumerations](../../../csharp/language-reference/keywords/enum.md)  
-  
- Structs fall into these categories:  
-  
--   Numeric types  
-  
-    -   [Integral types](../../../csharp/language-reference/keywords/integral-types-table.md)  
-  
-    -   [Floating-point types](../../../csharp/language-reference/keywords/floating-point-types-table.md)  
-  
--   [bool](../../../csharp/language-reference/keywords/bool.md)  
-  
--   User defined structs.  
+# Value types (C# Reference)
+
+The value types consist of two main categories:
+
+- [Structs](struct.md)
+
+- [Enumerations](enum.md)
+ 
+ Structs fall into these categories:
+
+- Numeric types and [char](char.md)
```

Comment:
I think it is odd to mention char twice, particularly in the title. "Numeric types and char" doesn't seem like a category to me. I'd stick with "Numeric types". Char is either a numeric type of not.

On the topic of numeric types, I am still not convinced that char is a numeric type. This is the best code I can write that makes me consider it as numeric.

```csharp
char ten = (char)10;
char twenty = (char)20;
Console.WriteLine(ten * twenty);
```

The first two files rely on an explicit cast. If it was implicit, I'd feel more compelled. The thing that makes it numeric type is that it uses ints as both a backing data structure and it has a well-defined conversion with ints. So, to my mind, calling it numeric is more academic than practical.

I think that characterizing structs is difficult. Are ValueTask and Span in the user-defined structs category? If so, then I'd switch your categorization of structs to be: primitive, special , and user-defined. "special" is a bad name but I cannot think of something better ATM. Span is neither primitive or user-defined, in my mind. Another framing might be: provided (as in "in-box") and user-defined.

Reply:
@richlander 
> I think that characterizing structs is difficult. 

Definitely. Thanks a lot for your response. It made me look at the article more critically.

I think the struct categorization is not necessary at all here. How does it help to reason about value types? What I've noticed is that the current "categorization" is about simple types and all other structs, which are called user-defined structs (yes, even [BigInteger](https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger?view=netframework-4.7.2) becomes a user-defined struct). So, I've removed the categorization and, instead, provided the definition (through enumeration) of the *simple types*. Why is it necessary to have such a category: simple types? Because those differ from all other structs and those differences are then in the article.

I've updated the article based on the [C# language spec](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/types#simple-types) and omitted the last "feature" of the simple types. Please let me know if you think that it should go to the docs as well.